src/lib/containers/access_requirement_list/AccessRequirementList.tsx

```diff
@@ -150,17 +155,21 @@ export default function AccessRequirementList({
             entityId,
           )
           const sortedAccessRequirements =
-            await sortAccessRequirementByCompletion(requirements)
-          setAccessRequirements(sortedAccessRequirements)
+            await sortAccessRequirementByCompletion(accessToken, requirements)
+          if (!isCancelled) {
+            setAccessRequirements(sortedAccessRequirements)
+          }
         } else {
           const sortedAccessRequirements =
-            await sortAccessRequirementByCompletion(accessRequirementFromProps!)
-          setAccessRequirements(sortedAccessRequirements)
+            await sortAccessRequirementByCompletion(
+              accessToken,
+              accessRequirementFromProps,
+            )
+          if (!isCancelled) {
+            setAccessRequirements(sortedAccessRequirements)
+          }
         }
 
-        const userProfile = await SynapseClient.getUserProfile(accessToken)
-        setUser(userProfile)
```

Comment:
Leave fetching the userProfile to react-query (should at least be removed from this hook, because fetching the profile does not need to be blocked by fetching Access Requirements)